src/qibo/core/terms.py

```diff
@@ -0,0 +1,273 @@
+import sympy
+from qibo import gates, K
+from qibo.config import raise_error, EINSUM_CHARS
+
+
+class HamiltonianTerm:
+    """Single term in a :class:`qibo.core.hamiltonians.SymbolicHamiltonian`.
+
+    This Hamiltonian is represented by a sum of such terms which are held in
+    its ``.terms`` attribute which is a list.
+
+    Args:
+        matrix (np.ndarray): Full matrix corresponding to the term representation
+            in the computational basis. Has size (2^n, 2^n) where n is the
+            number of target qubits of this term.
+        q (int): Target qubit ids.
```

Comment:
This is a list of ints, right?

Reply:
From the code side `q` is interpreted as a list of ints, but in terms of API the ints are just passed as seperate arguments because `*q` is used in definition, eg.:
```Python
HamiltonianTerm(matrix, 0, 1, 2, 3)
```
So I am not sure what is the proper way of implementing this. It is the same API with gates with multiple targets (eg. the `gates.Unitary` or `gates.I`).

Reply:
I think you can safely say list of ints.